x-pack/plugins/index_management/public/application/sections/home/data_stream_list/edit_data_retention_modal/edit_data_retention_modal.tsx

```diff
@@ -95,49 +68,46 @@ const configurationFormSchema: FormSchema = {
     validations: [
       {
         validator: ({ value, formData, customData }) => {
-          // If infiniteRetentionPeriod is set, we dont need to validate the data retention field
-          if (formData.infiniteRetentionPeriod) {
-            return undefined;
-          }
-
-          // If project level data retention is enabled, we need to enforce the global max retention
-          const { globalMaxRetention, enableProjectLevelRetentionChecks } = customData.value as any;
-          if (enableProjectLevelRetentionChecks) {
-            const currentValue = `${value}${formData.timeUnit}`;
-            if (globalMaxRetention && isRetentionBiggerThan(currentValue, globalMaxRetention)) {
-              return {
-                message: i18n.translate(
-                  'xpack.idxMgmt.dataStreamsDetailsPanel.editDataRetentionModal.dataRetentionFieldMaxError',
-                  {
-                    defaultMessage:
-                      'Maximum data retention period on this project is {maxRetention} days.',
-                    // Remove the unit from the globalMaxRetention value
-                    values: { maxRetention: globalMaxRetention.slice(0, -1) },
-                  }
-                ),
-              };
+          // We only need to validate the data retention field if infiniteRetentionPeriod is set to false
+          if (!formData.infiniteRetentionPeriod) {
+            // If project level data retention is enabled, we need to enforce the global max retention
+            const { globalMaxRetention, enableProjectLevelRetentionChecks } =
+              customData.value as any;
```

Comment:
Why is `any` used here? can it be avoided?

Reply:
I think this is added to avoid a type error because the type of `customData.value` is `unknown` (see [here](https://github.com/elastic/kibana/blob/92b145bcd739af9e132be158c08ba2f3c8cd43d1/src/plugins/es_ui_shared/static/forms/hook_form_lib/types.ts#L264)).